constrained-voucher.mkd

```diff
@@ -520,6 +506,92 @@ One key difference is that the names of the leaves in the YANG does not have a m
+The details of how it becomes an IDevID is defined in {{ieee802-1AR}}.
+While the Pledge device needs to have an IDevID object, it does not need to manipulate it.
+It can regard the IDevID as an opaque object which is placed into a DTLS setup.
+
+The IDevID certificate will have an associated private key.
+The Pledge will need to use that private key for the DTLS connection, and for signing of the voucher-request.
+However, none of these operations require any kind of manipulation of certificates.
+The private key can be loaded in whatever format is appropriate and efficient for the hardware.
+(EDNOTE: are there lwig references that would make sense here?)
+
+## The Registrar Trust Anchor
+
+When the Pledge first connects to the Registrar, the connection to the Registrar is provisional, as explained in section 5.6.2 of {{I-D.ietf-anima-bootstrapping-keyinfra}}.
+The Registrar provides its public key in a TLSServerCertificate, and the Pledge uses that to validate that integrity of the (D)TLS connection, but it does not validate the identity of
+the provided certificate.
+
+If instead of using a (TLSServer)Certificate of type X509 (see section 4.4.2 of {{RFC8446}}),
+a RawPublicKey object is used, then the Pledge does not need to any decoding of the RF5280 format certifiate to get the public key.
+
+## The Voucher Request
+
+The Pledge can put the Registrar's public key into the proximity-registrar-subject-public-key-info
+(PRSPKI) field of the voucher-request.
+(The proximity-registrar-sha256-of-subject-public-key-info can also be used if the 32-bytes of hash turned out to be smaller than a typical ECDSA key.)
+
+As the format of the PRSPKI field is identical to the TLS Certificate RawPublicKey, no manipulation at all is needed to insert this into a voucher-request.
+
+## The Voucher Reply
+
+A returned voucher will have a pinned-domain-subject-public-key-info field with the identical key as was found in the proximity field above, as well as in the TLS connection.
+Validation of this is what takes the connection out of the provisional state (see {{I-D.ietf-anima-bootstrapping-keyinfra}} section 5.6.2).
+
+The voucher needs to be validated first.
+The Pledge needs to have a public key to validate the signature from the MASA on the voucher.
+
+Certificate chains may be included in the COSE-signed Constrained Vouchers described in this document, using the x5bag attribute from {{I-D.ietf-cose-x509}}.
+However, this would require that the Pledge is able to form a certificate chain and validate it against a built-in Trust Anchor.
+
+A simpler method is that the MASA would sign the Voucher directly with a key pair for which the Pledge has the public key built-in to its firmware.
+That is, the MASA would use a degenerate PKI of level one.
+
+This has significant advantages on the Pledge, but presents security issues for the MASA, as it needs to keep a private online for signing purposes.
```

Comment:
From here on the content seems mostly security considerations; and these should be moved to that section. (Peter already commented something similar.)